arbor/fvm_layout.cpp

```diff
@@ -684,29 +684,38 @@ fvm_mechanism_data fvm_build_mechanism_data(const cable_cell_global_properties&
 
     struct synapse_instance {
         size_type cv;
-        std::map<std::string, double> param_value; // uses ordering of std::map
+        std::vector<std::pair<unsigned, double>> param_value; // sorted vector of <param_id,value> pairs
         size_type target_index;
     };
 
     for (const auto& entry: cell.synapses()) {
         const std::string& name = entry.first;
         mechanism_info info = catalogue[name];
-        std::map<std::string, double> default_param_value;
         std::vector<synapse_instance> sl;
 
+        // Map from a param string identifier to a unique unsigned int identifier
+        std::map<std::string, unsigned> param_map;
```

Comment:
A hash table (`std::unordered_map`) might be a better choice here, because below on line 715 the `map::at` call will do a binary search using string comparison, whereas with a hash table it will use a hash of the string.

Reply:
A pretty minor optimization, given that it isn't the pain point that you are optimizing for here, so let me know if you want to ignore it.

Reply:
No, that's a good point. I'll fix it :)

Reply:
Actually, I remembered that it was originally `unordered_map` but that broke the assertions at `fvm_layout.cpp:500` and `fvm_layout.cpp:501`. I don't think those assertions are necessary but I'm not comfortable removing them.

Reply:
:+1: